src/treemodel/selection.js

```diff
@@ -135,8 +156,44 @@ export default class Selection {
 	}
 
 	/**
+	 * Returns the first range in the selection. First range is the one which {@link core.treeModel.Range#start start} position
```

Comment:
Does it mean that ranges aren't sorted upon adding them? It sounds like it could be better to:

* sort them when adding,
* require that they are added in the right order.

Or is it caused by the fact that live ranges are used, so they may change their order after being added to the selection?